.gitlab/build_lassen.yml

```diff
@@ -37,5 +37,7 @@ clang_upstream_gfortran (PR build_on_lassen):
 clang_upstream_gfortran (Main build_with_deps_on_lassen):
   variables:
     COMPILER: "clang@upstream_gfortran"
-    SPEC: "@develop%${COMPILER}"
+    SPEC: "\"@develop+cuda%${COMPILER} cuda_arch=70\""
+    # The "-cuda" at the end of the host-config needs to be hardcoded for now
+    HOST_CONFIG: "lassen-blueos_3_ppc64le_ib_p9-${COMPILER}-cuda.cmake"
```

Comment:
It was on purpose that the host-config file name did not appear in CI.

Indeed it’s much easier to change the configuration/spec when not mentioning the host-config file name, which is "generated" at an intermediate step.

To do so, I simply made the assumption that the only host-config file present at the root directory was the one I want, since it has been produced in the "dependencies" step.

I hope it makes sense.

Reply:
that is true and a good point if these scripts are only used in one thing but the being that they are used to unify all CI, building public libraries, and even some development.  They need to work in the same way across all of those workflows.

You wouldn't want a script like this "finding" a host-config in the root that you didn't expect it to. when really you wanted it to test all of the host-configs for this platform.

Reply:
Allow me to insist:
in `.build_with_deps_blueos_3_ppc64le_ib_p9_script:`
The script calls are
```
    - lalloc 1 -W 20 ${LSF_ACCOUNT} scripts/llnl/build_tpls.py --spec=${SPEC} --directory=${CI_PROJECT_NAME}
    - lalloc 1 -W 10 ${LSF_ACCOUNT} scripts/llnl/build_src.py
```
So it seems to me `HOST_CONFIG` is not used (or by environment variable?), otherwise wouldn’t it be with `--host-config ${HOST_CONFIG}` in the second call?

I still think that _even outside of CI_ it would be useful to be able to build the code and its dependencies with a single command and only the "spec" and _not_ the host-config file name.

Reply:
It is used by environment variable here, but it makes sense to specify it explicitly - I'll add that in.